service/src/test/java/uk/gov/hmcts/reform/fpl/utils/JudgeAndLegalAdvisorHelperTest.java

```diff
@@ -97,5 +101,85 @@ private void shouldExtractOtherTitleDescriptionWhenOtherTitleSelected() {
 
         assertThat(magistrateName).isEqualTo("His Excellency John Doe");
     }
+
+    @Test
+    void shouldReturnAllocatedJudgeWhenUseAllocatedJudgeSelected() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = JudgeAndLegalAdvisor.builder()
+            .judgeTitle(MAGISTRATES)
+            .judgeLastName("Stevenson")
+            .legalAdvisorName("John Papa")
+            .useAllocatedJudge("Yes")
+            .allocatedJudgeLabel("Case assigned to: His Honour Judge Dread")
+            .build();
+
+        Judge allocatedJudge = buildJudge();
+
+        JudgeAndLegalAdvisor expectedJudge = getSelectedJudge(judgeAndLegalAdvisor, allocatedJudge);
+
+        assertThat(expectedJudge.getJudgeTitle()).isEqualTo(HIS_HONOUR_JUDGE);
+        assertThat(expectedJudge.getJudgeLastName()).isEqualTo("Dread");
+        assertThat(expectedJudge.getLegalAdvisorName()).isEqualTo("John Papa");
+    }
+
+    @Test
+    void shouldReturnInputtedJudgeWhenUseAllocatedJudgeNotSelected() {
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = JudgeAndLegalAdvisor.builder()
+            .judgeTitle(MAGISTRATES)
+            .judgeLastName("Stevenson")
+            .legalAdvisorName("John Papa")
+            .useAllocatedJudge("No")
+            .allocatedJudgeLabel("Case assigned to: His Honour Judge Dread")
+            .build();
```

Comment:
As we use this here & the test above we could extract to private method with parameter useAllocatedJudge set to yes/no depending. 

- if you don't wish to refactor here I don't mind